src/sidebar/util/fetch-config.js

```diff
@@ -75,3 +97,87 @@ export function fetchConfig(appConfig, window_ = window) {
     return mergedConfig;
   });
 }
+
+/**
+ * Merge client configuration from h service with config fetched from
+ * an embedding frame synchronously.
+ *
+ * Use this method to retrieve the config synchronously from the embedding
+ * frame's query string. See tests for more details.
+ *
+ * @param {Object} appConfig - App config settings rendered into `app.html` by the h service.
+ * @param {Object} hostPageConfig - App configuration specified by the embedding  frame.
+ * @return {Object} - The merged settings.
+ */
+function fetchConfigEmbed(appConfig, hostPageConfig) {
+  const mergedConfig = {
+    ...appConfig,
+    ...hostPageConfig,
+  };
+  mergedConfig.apiUrl = getApiUrl(mergedConfig);
+  return mergedConfig;
+}
+
+/**
+ * Merge client configuration from h service with config fetched from
+ * an embedding frame asynchronously.
```

Comment:
```suggestion
 * a parent frame asynchronously.
```

The frame supplying the config might not be the frame that directly embeds the client, and isn't in the lms case.